api/v1beta1/types_template.go

```diff
@@ -90,6 +90,7 @@ type VnetPeeringsTemplateSpec []VnetPeeringClassSpec
 
 // SubnetTemplateSpec specifies a template for a subnet.
 type SubnetTemplateSpec struct {
+
 	SubnetClassSpec `json:",inline"`
```

Comment:
I think we wanted to introduce validation of the names in AzureClusterTemplate 

Should be ca. https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/2097e3f741a173f7713f5ccf523b2d3dd906ac57/api/v1beta1/azurecluster_validation.go#L176-L182